Sources/Rendering/WebGPU/Types/index.js

```diff
@@ -0,0 +1,74 @@
+// ----------------------------------------------------------------------------
+// vtkWebGPUDevice static functions
+//
+// WebGPU uses types in a many places and calls, and often those types
+// need to be associated with byte sizes, alignments, native arrays etc.
+// The folowing methods are designed to help vtk.js introspect those types.
+// WebGPU currently tends to use multiple type formats:
+//  - buffer types such as float32x4
+//  - shader types suchs as vec4<f32>
+//  - texture types such as rgba32float
+// ----------------------------------------------------------------------------
+
+function getByteStrideFromBufferFormat(format) {
+  if (!format) return 0;
+  let numComp = 1;
+  if (format.substring(format.length - 2) === 'x4') numComp = 4;
```

Comment:
could this be replaced by `const numComp = Number(format.split('x').pop());` ?
It will be less efficient if called often though.